lib/chef/resource/resource_notification.rb

```diff
@@ -26,7 +26,11 @@ class Notification
 
       def initialize(resource, action, notifying_resource)
         @resource = resource
-        @action = action
+        @action = if action.respond_to?(:to_sym)
+                    action.to_sym
+                  else
+                    action
+                  end
```

Comment:
you can just call action.to_sym here.  Symbol#to_sym is just a no-op

Reply:
That's what I first did. Then some unit tests failed because action was set to nil in some of them. So I decided to check if value of action is actually supports to_sym

Reply:
You can use `action&.to_sym`.

Reply:
Oh, thank you. Will do.